# **MMORPG 거래소의 거래정보를 통한 게임화폐가치 예측**



정원희

한양대학교 파이낸스경영학과

---

# 목차


>[서론](#scrollTo=LKdoy_YPY57N)

>[가설 설정](#scrollTo=WQccEyn5mj1l)

>[데이터 수집](#scrollTo=lXW5MGzlpVjp)

>[데이터 분석](#scrollTo=TLiXmPiQOKg0)

>[4-1. 시나리오1](#scrollTo=KsjndhOuboww)

>[4-2. 시나리오2](#scrollTo=PgVwSlJ8cFvD)

>[4-3. 시나리오3](#scrollTo=jEySB5wfiljk)

>[4-4. 시나리오4](#scrollTo=MO2JzRdmkApm)

>[4-5. 시나리오5](#scrollTo=cICxEpM5l2KJ)

>[결론](#scrollTo=BxHI556FoIpr)



# 1. 서론




*   온라인 게임, 특히 MMORPG(Massively Multiplayer Online Role-Playing Game)는 **수많은 유저들이 인터넷을 통해 주어진 역할을 수행하는 게임**을 말한다.

*   유저들은 게임공간 안에서 **몬스터를 사냥해서 전리품과 화폐를 획득하며, 이를 다른 사람들과 같이 거래할 수 있다.**

*   게임개발사는 유저들의 플레이방식을 분석하여 게임경제가 적절하게 유지되도록 **게임화폐가치 및 게임재화가치를 조절하는 역할을 수행**한다.

*   일반적으로 MMORPG는** '거래소'라는 시스템이 있어서 유저들은 일정 수수료를 지불하고 자유롭게 자신의 아이템을 거래할 수 있다.**

*   이와 다르게 게임 내 시장이 아닌** 게임 외적인 공간에서 아이템을 거래하는 시장이 존재하는데, '아이템 거래 사이트'라고 한다.**



***게임경제와 실물경제의 공통점***
   
   
   

■   MMORPG는 하나의 독립적인 공간, 시장경제체제가 이루어지는 사회이다.1)

![대체 텍스트](https://postfiles.pstatic.net/MjAxODA2MjFfMTgg/MDAxNTI5NTY2MjIxMjkw.KMWcBNH5-nHSPGzvr0wSd1Ly57p_Kfy6R4DRhMiaGMQg.EjVRA_BRJJ868HoZ5-gLz4FWc_nC0FSUoyQKN72Txngg.JPEG.bububu111/noname01.jpg?type=w773)

>* 경제활동인구 ≒ 게임이용자

>* 국가                 ≒ 게임

>* 법률                 ≒ 게임시스템

>* 생산&소비      ≒ 게임아이템

>* 화폐                 ≒ 게임화폐

>* 정부                 ≒ 게임개발사

>* 경제                 ≒ 게임경제

>* 경제 정책        ≒ 게임 업데이트 안

■   게임 접속을 종료하더라도 게임이 계속 진행된다.

>* MMORPG는 게임 이용자의 플레이 여부에 상관없이 진행된다.

>* 마치 현실세계에서 잠을 자고있는 동안에도 세상이 돌아가듯이, MMORPG에서도 다른 유저들의 행동 하나하나가 전부 서버에 기록되며 저장된다.

---

1) 신범수(2008), MMORPG의 거시경제모델 연구

***게임경제와 실물경제의 차이점***

■   게임경제는 최저생계비가 필요하지 않다.

>* 실물경제에서 사람들은 기본적인 최저생계비가 필요하다(주거비용, 식비 등) 

>* 하지만 게임경제에서는 최저생계비가 없기때문에 **게임개발사가 따로 화폐 회수 시스템을
구축하지 않으면 필연적으로 인플레이션이 발생할 수 밖에 없다.**

>* 따라서 게임개발사는 게임경제 및 인플레이션 관리를 위한 여러가지 화폐 회수 컨텐츠를 개발하고 관리한다.(e.g 장비 수리비, 소모품 비용, 강화비용 등)

■   게임경제는 PvE(Player versus Environment)가 주된 컨텐츠이다.

>* MMORPG는 게임서버에 설계된 게임환경에 의해 게임이 진행된다.

>* 필드에 생성되는 몬스터를 죽여서 아이템을 얻거나, 값비싼 전리품을 얻기 위해 위험한 던전에 들어가기도 한다.

>* PvP(Player versus Player)를 통해 보상을 얻기도 하지만 일반적으로 비주류 컨텐츠이다.

>* 게임 내 PvE의 비중이 커야 게임 개발자가 게임시스템을 통제하기 쉽기 때문이다.

>* 따라서 실물경제에 비해서 게임경제는 통제할 수 있는 범위가 굉장히 넓은 편이다.

■   치명적인 오류나 사고가 발생했을 때 유연하게 대처가 가능하다.

>* 게임 내 버그나 시스템 오류로 인한 피해가 발생했을 때 게임서버의 로그를 분석해서 문제 해결이 가능하다.

>* 하지만 실물경제는 재난 상황이나 국제경제의 영향에 의한 피해에 대한 민감도가 높다.


# 2. 가설 설정
* 게임 아이템을 거래할 수 있는 시장은 두 곳이 있다.

* 하나는 게임 내 시장인 '거래소'이고, 나머지는 게임 외 시장인 '아이템 거래 사이트'이다.

* **'거래소'는 게임 내에 존재하고 매우 손쉽게 아이템을 사고 팔 수 있는 장벽이 낮은 시장**이다.

*  따라서 '거래소'는 효율적인 시장으로 게임 아이템의 가격에 영향을 줄만한 정보(게임 업데이트 예정 공시, 아이템 밸런스 패치 등 )가 매우 빠르게 반영된다.

* 이에 비해서 '아이템 거래 사이트'는 거래하기 위한 절차가 까다롭고 복잡하여 장벽이 높은 시장이다.

* 즉, **'아이템 거래 사이트'는 가격에 대한 정보의 반영이 느린 시장**이다.

* 그런데 **두 시장 모두 같은 재화에 대한 거래가 이루어 지는 시장**이다.


◎  **정보 반영 속도 차이로 인해 '거래소'가 '아이템 거래 사이트'의 선행지표가 될 수 있다.**

◎  **'거래소'의 아이템 거래내역 데이터를 분석하면 '아이템 거래 사이트'내 거래 가격을 예측할 수 있을 것이다.** 







![대체 텍스트](https://postfiles.pstatic.net/MjAxODA2MjFfMzcg/MDAxNTI5NTY2OTg4MDI0.F9HkSptNe2bOjtjOBwtjoR11rFTfRGx93gO9MT7MEUsg.yB5Hic8b6gZMPVU-weKhOoyu2YzyllHXQkRN7A4fX6sg.JPEG.bububu111/ScreenShot2018_0621_164234493.jpg?type=w773)


                                      <  게임 내 거래소 모습 >



# 3. 데이터 수집

**○ 가격은 더 좋은 결과를 도출하기 위해 로그변환을 취했다.**





▼ 일별 게임화폐 시세

* 데이터 수집 대상 게임으로 국내 MMORPG 장르 1위 게임인 '던전앤파이터'를 선정하였다.

* 게임개발사에서 오픈 API를 제공하므로 '거래소'의 데이터를 분석하기 용이하기 때문이다.

* 게임화폐가치는 국내 아이템 거래 사이트가 제공한다.

*  그중에서도 양질의 데이터를 구하기 위해 아이템 거래량이 가장 높은 '아이템 매니아'라는 사이트
에서 제공하는 일별 게임화폐가치를 참고하였다.

* 게임안에 여러 서버가 존재하는데, 그중에서도 가장 유저수가 많은 '카인'서버를 기준으로 데이터를 수집하였다.

In [38]:
# '아이템 매니아' 사이트에서 제공하는 일별 게임화폐시세 자료


stock_file_name = 'https://www.dropbox.com/s/y5g7aj60nztnaz4/df_goldPrice.csv?dl=1'
encoding = 'korean' # 문자 인코딩
goldPrice = pd.read_csv(stock_file_name, encoding=encoding, names=['Date','Price'],header=None)
goldPrice

# 2008년 9월 1일 ~ 2018년 6월 20일까지의 일별 게임화폐가치
# Price 단위 : 1000만 골드당 원화가치

Date      Price
0     2018-06-20   9.034796
1     2018-06-19   9.050172
2     2018-06-18   9.053687
3     2018-06-17   9.079092
4     2018-06-16   9.104980
5     2018-06-15   9.097172
6     2018-06-14   9.099409
7     2018-06-13   9.106090
8     2018-06-12   9.103868
9     2018-06-11   9.110520
10    2018-06-10   9.159047
11    2018-06-09   9.179881
12    2018-06-08   9.152711
13    2018-06-07   9.129130
14    2018-06-06   9.209340
15    2018-06-05   9.219300
16    2018-06-04   9.192176
17    2018-06-03   9.197255
18    2018-06-02   9.222269
19    2018-06-01   9.221280
20    2018-05-31   9.203316
21    2018-05-30   9.174713
22    2018-05-29   9.180912
23    2018-05-28   9.182969
24    2018-05-27   9.213336
25    2018-05-26   9.275191
26    2018-05-25   9.281730
27    2018-05-24   9.235033
28    2018-05-23   9.075665
29    2018-05-22   9.097172
...          ...        ...
3544  2008-09-30  10.503971
3545  2008-09-29  10.521642
3546  2008-09-28  10.565918
3547  2008-09-27  10.563078
3548  2008-09-26  10.537415
3549  2008-09-25  10.531830
3550  2008-09-24  10.520024
3551  2008-09-23  10.536354
3552  2008-09-22  10.545078
3553  2008-09-21  10.572342
3554  2008-09-20  10.584815
3555  2008-09-19  10.588855
3556  2008-09-18  10.593630
3557  2008-09-17  10.606585
3558  2008-09-16  10.640891
3559  2008-09-15  10.618396
3560  2008-09-14  10.610291
3561  2008-09-13  10.612508
3562  2008-09-12  10.610044
3563  2008-09-11  10.618396
3564  2008-09-10  10.607327
3565  2008-09-09  10.560230
3566  2008-09-08  10.569264
3567  2008-09-07  10.656789
3568  2008-09-06  10.671510
3569  2008-09-05  10.687389
3570  2008-09-04  10.653251
3571  2008-09-03  10.599879
3572  2008-09-02  10.558154
3573  2008-09-01  10.554510

[3574 rows x 2 columns]

▼ 일별 게임 아이템 거래가격

* 유저들이 가장 많이 거래하는 재료아이템 두 가지(마그토늄, 데이터 칩)를 선정하였다.

* 두 아이템은 MMORPG의 주요 컨텐츠인 레이드(다수의 유저들이 강력한 몬스터를 잡기 위한 협동 임무)입장 시 필요한 아이템으로, 거래소에서 거래량이 가장 많은 아이템이다.

* 위 두 아이템의 거래소 데이터를 api로 불러와서 거래량과 가격을 기록한다. 

* 과거부터 이 데이터를 모아서 일별 게임 아이템 거래가격을 산출한다.

In [71]:
# API를 활용한 던전앤파이터 거래소 아이템 데이터 크롤링   (마그토늄)


import pandas as pd
import numpy as np
from bs4 import BeautifulSoup
import requests
import types
from types import *


url = 'https://api.neople.co.kr/df/auction?limit=400&itemName=%eb%a7%88%ea%b7%b8%ed%86%a0%eb%8a%84&sort=unitPrice:asc&limit=%3Climit%3E&wordType=match&apikey=JBIBYEzdSbwZF3k4QQRWcM5jnhDwCYSn'  #오픈api를 통해 거래소 데이터 요청
req = requests.get(url)
html = req.text
soup = BeautifulSoup(html, 'html.parser')



def is_start(num):
  while html[num] not in '1234567890':
    num += 1
  return num

def is_end(num):
  while html[num] != ',':
    num += 1
  return num


Date = []
Vol = []
Price = []

check = 0


while html.find("auctionNo") != -1: # 경매장 맨 마지막 리스트까지 검색
  date = html.find("regDate") # 경매장에 아이템을 등록하는 시기
  t = str(html[(date+10):(date+29)])
  Date.append(t)


  count = html.find("count") # 등록된 아이템의 개수
  v = int(html[is_start(count):is_end(count)])
  Vol.append(v)


  price = html.find("unitPrice") # 등록된 아이템의 개당 가격
  p = int(html[is_start(price):is_end(price)])
  Price.append(p)



  check = int(price+10)

  html = html[check:]



df = { # Date, Volum, UnitPrice로 데이터 정렬
    'Date' : Date,
    'Volume' : Vol,
    'UnitPrice' : Price
}

dataFrame = pd.DataFrame(df)
dataFrame

Date  UnitPrice  Volume
0    2018-06-21 18:06:06        887     952
1    2018-06-21 18:04:29        888     354
2    2018-06-21 18:05:45        888      24
3    2018-06-21 18:05:48        888    1135
4    2018-06-21 18:05:52        888     337
5    2018-06-21 18:04:44        889      33
6    2018-06-21 18:03:44        900     198
7    2018-06-21 18:04:38        920     136
8    2018-06-21 18:01:55        930      65
9    2018-06-21 18:04:18        940    4290
10   2018-06-21 18:01:52        945     863
11   2018-06-21 18:00:17        949       1
12   2018-06-21 17:53:10        950      10
13   2018-06-21 18:01:21        964     891
14   2018-06-21 17:56:52        965      67
15   2018-06-21 17:57:52        965      71
16   2018-06-21 17:50:26        967      97
17   2018-06-21 18:00:17        969     800
18   2018-06-21 17:24:03        970       6
19   2018-06-21 17:43:46        970       7
20   2018-06-21 17:44:59        970      12
21   2018-06-21 17:50:08        970     172
22   2018-06-21 17:56:07        970      11
23   2018-06-21 17:57:44        970      45
24   2018-06-21 17:58:23        970     771
25   2018-06-21 17:59:03        970    2420
26   2018-06-21 17:59:07        970      13
27   2018-06-21 17:59:56        970      45
28   2018-06-21 18:00:53        970      39
29   2018-06-21 18:04:40        970    1120
..                   ...        ...     ...
176  2018-06-20 19:44:48       1199   10000
177  2018-06-20 23:04:19       1200     444
178  2018-06-21 13:04:35       1200    1209
179  2018-06-21 14:15:30       1200   30183
180  2018-06-20 19:55:37       1299   14661
181  2018-06-20 18:13:34       1310     123
182  2018-06-20 19:52:07       1500    4732
183  2018-06-20 23:23:42       1500    9757
184  2018-06-20 20:42:29       1550   10000
185  2018-06-20 23:21:58       1800     595
186  2018-06-21 04:47:52       1999   40871
187  2018-06-21 00:19:23       2000       6
188  2018-06-21 00:10:16       2800     989
189  2018-06-21 01:41:41       2800     113
190  2018-06-21 03:14:57       2800     118
191  2018-06-21 03:30:21       2800     114
192  2018-06-21 12:30:20       2800     494
193  2018-06-21 13:32:06       2997   22088
194  2018-06-20 20:57:47       3000   15000
195  2018-06-21 00:15:26       5000    9209
196  2018-06-21 10:06:19       5000   25428
197  2018-06-20 22:53:01       8980     172
198  2018-06-20 18:23:44       9111      10
199  2018-06-20 21:31:14       9950    1000
200  2018-06-20 22:45:07       9952     100
201  2018-06-20 23:32:00      10000    1642
202  2018-06-21 00:38:26      91007      21
203  2018-06-21 00:24:18      91111      10
204  2018-06-21 11:05:10      97011       1
205  2018-06-20 21:46:09   11100000       7

[206 rows x 3 columns]

* 위 방식과 같이 오픈 api를 통해 과거부터 거래소 데이터를 받아서 아이템의 거래가격을 도출하고 싶었다.

* 하지만 **문의 결과 오픈 api는 '현재 거래소 데이터'만 제공**하기 때문에 현재부터 데이터를 축적하는 방법밖에 없다.

* 그래서** 일반 유저가 약 한달 간 서버에 저장해놓은 아이템 거래 데이터를 가져오는 대안을 마련**했다.

In [40]:
# 일별 데이터 칩 아이템 시세

stock_file_name2 = 'https://www.dropbox.com/s/wwnwwkzt6n3lkua/df_chip.csv?dl=1'
encoding = 'korean' # 문자 인코딩
chipPrice = pd.read_csv(stock_file_name2, encoding=encoding, names=['Date','Price'],header=None)
chipPrice

Date     Price
0   2018-06-21  8.856234
1   2018-06-20  8.866582
2   2018-06-19  8.848653
3   2018-06-18  8.849371
4   2018-06-17  8.875567
5   2018-06-16  8.811652
6   2018-06-15  8.747193
7   2018-06-14  8.695841
8   2018-06-13  8.721276
9   2018-06-12  8.650850
10  2018-06-11  8.694502
11  2018-06-10  8.701679
12  2018-06-09  8.628913
13  2018-06-08  8.575273
14  2018-06-07  8.535622
15  2018-06-06  8.573384
16  2018-06-05  8.637817
17  2018-06-04  8.682369
18  2018-06-03  8.653122
19  2018-06-02  8.613775
20  2018-06-01  8.532870
21  2018-05-31  8.537192
22  2018-05-30  8.582232
23  2018-05-29  8.614683
24  2018-05-28  8.621553
25  2018-05-27  8.676587
26  2018-05-26  8.688622
27  2018-05-25  8.647871
28  2018-05-24  8.705662
29  2018-05-23  8.737613
30  2018-05-22  8.762646
31  2018-05-21  8.782323
32  2018-05-20  8.805675
33  2018-05-19  8.765615
34  2018-05-18  8.705165

In [41]:
# 일별 마그토늄 아이템 시세

stock_file_name3 = 'https://www.dropbox.com/s/ap07gvfshl6fua1/df_mag.csv?dl=1'
encoding = 'korean' # 문자 인코딩
magPrice = pd.read_csv(stock_file_name3, encoding=encoding, names=['Date','Price'],header=None)
magPrice

Date     Price
0   2018-06-21  6.844815
1   2018-06-20  6.814543
2   2018-06-19  6.848005
3   2018-06-18  6.879356
4   2018-06-17  6.913737
5   2018-06-16  6.915723
6   2018-06-15  6.900731
7   2018-06-14  6.892642
8   2018-06-13  6.857514
9   2018-06-12  6.842683
10  2018-06-11  6.897705
11  2018-06-10  6.948897
12  2018-06-09  6.943122
13  2018-06-08  6.858565
14  2018-06-07  6.891626
15  2018-06-06  6.906755
16  2018-06-05  6.883463
17  2018-06-04  7.003065
18  2018-06-03  7.008505
19  2018-06-02  6.978214
20  2018-06-01  6.913737
21  2018-05-31  6.911747
22  2018-05-30  6.924612
23  2018-05-29  6.882437
24  2018-05-28  7.009409
25  2018-05-27  7.085064
26  2018-05-26  7.113956
27  2018-05-25  7.090077
28  2018-05-24  7.172425
29  2018-05-23  7.191429
30  2018-05-22  7.163947
31  2018-05-21  7.228388
32  2018-05-20  7.252054
33  2018-05-19  7.190676
34  2018-05-18  7.122060

# 4. 데이터 분석

* 수집한 데이터를 통해 가설을 검증하기 위해 회귀분석을 실시하고자 한다.

* 가설을 통해 도출해낸 모델은 다음과 같다.


![대체 텍스트](https://postfiles.pstatic.net/MjAxODA2MjFfMTI0/MDAxNTI5NTY2ODQ3Nzkz.DTQrrebGhPs8DXWr4EIKgX_OwfKtv3KpqI9m4DDqClcg.fEVPoLzRAftYdav2fhGl8hUwLedCXFCFIV9kPfIlPj8g.PNG.bububu111/%EC%88%98%EC%8B%9D.PNG?type=w773)


y = 일별 게임화페시세

mprice = 일별 마그토늄 아이템 시세

dprice = 일별 데이터칩 아이템 시세


* 여기서 마그토늄 시세와 데이터칩 시세의 t시점을 각각 t-1, t-2로 변형시켜서 어떤 시나리오가 가장 설명력이 높은지 책정한다.




# 4-1. 시나리오1

y = mprice(t) + dprice(t)의 경우

In [61]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import statsmodels.formula.api as sm
from statsmodels.sandbox.regression.predstd import wls_prediction_std
import statsmodels.stats.sandwich_covariance as sw
import statsmodels.stats.api as sms
import statsmodels.stats.stattools as smt
from statsmodels.compat import lzip
from arch import arch_model


set1 = { 
    'Date' : chipPrice.loc[:,'Date'],
    'goldPrice': goldPrice.loc[0:34,'Price'],
    'magPrice' : magPrice.loc[:,'Price'],
    'chipPrice' : chipPrice.loc[:,'Price'] 
}

data1 = pd.DataFrame(set1)

data1

/usr/local/lib/python2.7/dist-packages/statsmodels/compat/pandas.py:56: FutureWarning: The pandas.core.datetools module is deprecated and will be removed in a future version. Please use the pandas.tseries module instead.
  from pandas.core import datetools


Date  chipPrice  goldPrice  magPrice
0   2018-06-21   8.856234   9.034796  6.844815
1   2018-06-20   8.866582   9.050172  6.814543
2   2018-06-19   8.848653   9.053687  6.848005
3   2018-06-18   8.849371   9.079092  6.879356
4   2018-06-17   8.875567   9.104980  6.913737
5   2018-06-16   8.811652   9.097172  6.915723
6   2018-06-15   8.747193   9.099409  6.900731
7   2018-06-14   8.695841   9.106090  6.892642
8   2018-06-13   8.721276   9.103868  6.857514
9   2018-06-12   8.650850   9.110520  6.842683
10  2018-06-11   8.694502   9.159047  6.897705
11  2018-06-10   8.701679   9.179881  6.948897
12  2018-06-09   8.628913   9.152711  6.943122
13  2018-06-08   8.575273   9.129130  6.858565
14  2018-06-07   8.535622   9.209340  6.891626
15  2018-06-06   8.573384   9.219300  6.906755
16  2018-06-05   8.637817   9.192176  6.883463
17  2018-06-04   8.682369   9.197255  7.003065
18  2018-06-03   8.653122   9.222269  7.008505
19  2018-06-02   8.613775   9.221280  6.978214
20  2018-06-01   8.532870   9.203316  6.913737
21  2018-05-31   8.537192   9.174713  6.911747
22  2018-05-30   8.582232   9.180912  6.924612
23  2018-05-29   8.614683   9.182969  6.882437
24  2018-05-28   8.621553   9.213336  7.009409
25  2018-05-27   8.676587   9.275191  7.085064
26  2018-05-26   8.688622   9.281730  7.113956
27  2018-05-25   8.647871   9.235033  7.090077
28  2018-05-24   8.705662   9.075665  7.172425
29  2018-05-23   8.737613   9.097172  7.191429
30  2018-05-22   8.762646   9.083643  7.163947
31  2018-05-21   8.782323   9.096051  7.228388
32  2018-05-20   8.805675   9.112728  7.252054
33  2018-05-19   8.765615   9.108308  7.190676
34  2018-05-18   8.705165   9.102755  7.122060

In [55]:
result1 = sm.ols(formula = 'goldPrice ~ magPrice + chipPrice', data = data1).fit()

result1.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              goldPrice   R-squared:                       0.523
Model:                            OLS   Adj. R-squared:                  0.494
Method:                 Least Squares   F-statistic:                     17.57
Date:                Thu, 21 Jun 2018   Prob (F-statistic):           7.08e-06
Time:                        08:18:08   Log-Likelihood:                 59.315
No. Observations:                  35   AIC:                            -112.6
Df Residuals:                      32   BIC:                            -108.0
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     12.7002      0.781     16.258      0.000      11.109      14.291
magPrice       0.0856      0.063      1.363      0.182      -0.042       0.213
chipPrice     -0.4772      0.081     -5.895      0.000      -0.642      -0.312
==============================================================================
Omnibus:                        4.402   Durbin-Watson:                   0.855
Prob(Omnibus):                  0.111   Jarque-Bera (JB):                3.086
Skew:                           0.688   Prob(JB):                        0.214
Kurtosis:                       3.473   Cond. No.                     1.12e+03
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.12e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

**결과요약**

* magPrice의 p-value는 0.182로 설명력이 없었다.
* 하지만 chipPrice의 p-value는 0.000으로 설명력이 있었다.
* **durbin watson 결과 0.85가 나왔다.**

# 4-2. 시나리오2
y = mprice(t) + mprice(t-1) +  dprice(t) + dprice(t-1)의 경우

In [65]:
set2 = { 
    'Date' : chipPrice.loc[1::,'Date'],
    'goldPrice': goldPrice.loc[1:34,'Price'],
    'magPrice' : magPrice.loc[1:,'Price'],
    'magPrice1': magPrice.loc[1:,'Price'],
    'chipPrice' : chipPrice.loc[1:,'Price'],
    'chipPrice1' : chipPrice.loc[1:,'Price']
}

data2 = pd.DataFrame(set2)

data2

Date  chipPrice  chipPrice1  goldPrice  magPrice  magPrice1
1   2018-06-20   8.866582    8.866582   9.050172  6.814543   6.814543
2   2018-06-19   8.848653    8.848653   9.053687  6.848005   6.848005
3   2018-06-18   8.849371    8.849371   9.079092  6.879356   6.879356
4   2018-06-17   8.875567    8.875567   9.104980  6.913737   6.913737
5   2018-06-16   8.811652    8.811652   9.097172  6.915723   6.915723
6   2018-06-15   8.747193    8.747193   9.099409  6.900731   6.900731
7   2018-06-14   8.695841    8.695841   9.106090  6.892642   6.892642
8   2018-06-13   8.721276    8.721276   9.103868  6.857514   6.857514
9   2018-06-12   8.650850    8.650850   9.110520  6.842683   6.842683
10  2018-06-11   8.694502    8.694502   9.159047  6.897705   6.897705
11  2018-06-10   8.701679    8.701679   9.179881  6.948897   6.948897
12  2018-06-09   8.628913    8.628913   9.152711  6.943122   6.943122
13  2018-06-08   8.575273    8.575273   9.129130  6.858565   6.858565
14  2018-06-07   8.535622    8.535622   9.209340  6.891626   6.891626
15  2018-06-06   8.573384    8.573384   9.219300  6.906755   6.906755
16  2018-06-05   8.637817    8.637817   9.192176  6.883463   6.883463
17  2018-06-04   8.682369    8.682369   9.197255  7.003065   7.003065
18  2018-06-03   8.653122    8.653122   9.222269  7.008505   7.008505
19  2018-06-02   8.613775    8.613775   9.221280  6.978214   6.978214
20  2018-06-01   8.532870    8.532870   9.203316  6.913737   6.913737
21  2018-05-31   8.537192    8.537192   9.174713  6.911747   6.911747
22  2018-05-30   8.582232    8.582232   9.180912  6.924612   6.924612
23  2018-05-29   8.614683    8.614683   9.182969  6.882437   6.882437
24  2018-05-28   8.621553    8.621553   9.213336  7.009409   7.009409
25  2018-05-27   8.676587    8.676587   9.275191  7.085064   7.085064
26  2018-05-26   8.688622    8.688622   9.281730  7.113956   7.113956
27  2018-05-25   8.647871    8.647871   9.235033  7.090077   7.090077
28  2018-05-24   8.705662    8.705662   9.075665  7.172425   7.172425
29  2018-05-23   8.737613    8.737613   9.097172  7.191429   7.191429
30  2018-05-22   8.762646    8.762646   9.083643  7.163947   7.163947
31  2018-05-21   8.782323    8.782323   9.096051  7.228388   7.228388
32  2018-05-20   8.805675    8.805675   9.112728  7.252054   7.252054
33  2018-05-19   8.765615    8.765615   9.108308  7.190676   7.190676
34  2018-05-18   8.705165    8.705165   9.102755  7.122060   7.122060

In [66]:
result2 = sm.ols(formula = 'goldPrice ~ magPrice + magPrice1 + chipPrice + chipPrice1', data = data2).fit()

result2.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              goldPrice   R-squared:                       0.482
Model:                            OLS   Adj. R-squared:                  0.449
Method:                 Least Squares   F-statistic:                     14.43
Date:                Thu, 21 Jun 2018   Prob (F-statistic):           3.73e-05
Time:                        08:50:31   Log-Likelihood:                 57.303
No. Observations:                  34   AIC:                            -108.6
Df Residuals:                      31   BIC:                            -104.0
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     12.6283      0.800     15.793      0.000      10.997      14.259
magPrice       0.0385      0.033      1.182      0.246      -0.028       0.105
magPrice1      0.0385      0.033      1.182      0.246      -0.028       0.105
chipPrice     -0.2310      0.043     -5.368      0.000      -0.319      -0.143
chipPrice1    -0.2310      0.043     -5.368      0.000      -0.319      -0.143
==============================================================================
Omnibus:                        4.366   Durbin-Watson:                   0.850
Prob(Omnibus):                  0.113   Jarque-Bera (JB):                3.040
Skew:                           0.695   Prob(JB):                        0.219
Kurtosis:                       3.460   Cond. No.                     4.24e+18
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The smallest eigenvalue is 4.72e-34. This might indicate that there are
strong multicollinearity problems or that the design matrix is singular.
"""

**결과요약**

* magPrice의 p-value는 둘다 0.246으로 설명력이 없었다.
* 하지만 chipPrice의 p-value는 0.000으로 설명력이 있었다.
* **durbin watson 결과 0.85가 나왔다.**

# 4-3. 시나리오3
y = mprice(t) + mprice(t-1) + mprice(t-2) +  dprice(t) + dprice(t-1) + dprice(t-2)의 경우

In [67]:
set3 = { 
    'Date' : chipPrice.loc[2::,'Date'],
    'goldPrice': goldPrice.loc[2:34,'Price'],
    'magPrice' : magPrice.loc[2:,'Price'],
    'magPrice1': magPrice.loc[2:,'Price'],
    'magPrice2': magPrice.loc[2:,'Price'],    
    'chipPrice' : chipPrice.loc[2:,'Price'],
    'chipPrice1' : chipPrice.loc[2:,'Price'],
    'chipPrice2' : chipPrice.loc[2:,'Price']
}

data3 = pd.DataFrame(set3)

data3

Date  chipPrice  chipPrice1  chipPrice2  goldPrice  magPrice  \
2   2018-06-19   8.848653    8.848653    8.848653   9.053687  6.848005   
3   2018-06-18   8.849371    8.849371    8.849371   9.079092  6.879356   
4   2018-06-17   8.875567    8.875567    8.875567   9.104980  6.913737   
5   2018-06-16   8.811652    8.811652    8.811652   9.097172  6.915723   
6   2018-06-15   8.747193    8.747193    8.747193   9.099409  6.900731   
7   2018-06-14   8.695841    8.695841    8.695841   9.106090  6.892642   
8   2018-06-13   8.721276    8.721276    8.721276   9.103868  6.857514   
9   2018-06-12   8.650850    8.650850    8.650850   9.110520  6.842683   
10  2018-06-11   8.694502    8.694502    8.694502   9.159047  6.897705   
11  2018-06-10   8.701679    8.701679    8.701679   9.179881  6.948897   
12  2018-06-09   8.628913    8.628913    8.628913   9.152711  6.943122   
13  2018-06-08   8.575273    8.575273    8.575273   9.129130  6.858565   
14  2018-06-07   8.535622    8.535622    8.535622   9.209340  6.891626   
15  2018-06-06   8.573384    8.573384    8.573384   9.219300  6.906755   
16  2018-06-05   8.637817    8.637817    8.637817   9.192176  6.883463   
17  2018-06-04   8.682369    8.682369    8.682369   9.197255  7.003065   
18  2018-06-03   8.653122    8.653122    8.653122   9.222269  7.008505   
19  2018-06-02   8.613775    8.613775    8.613775   9.221280  6.978214   
20  2018-06-01   8.532870    8.532870    8.532870   9.203316  6.913737   
21  2018-05-31   8.537192    8.537192    8.537192   9.174713  6.911747   
22  2018-05-30   8.582232    8.582232    8.582232   9.180912  6.924612   
23  2018-05-29   8.614683    8.614683    8.614683   9.182969  6.882437   
24  2018-05-28   8.621553    8.621553    8.621553   9.213336  7.009409   
25  2018-05-27   8.676587    8.676587    8.676587   9.275191  7.085064   
26  2018-05-26   8.688622    8.688622    8.688622   9.281730  7.113956   
27  2018-05-25   8.647871    8.647871    8.647871   9.235033  7.090077   
28  2018-05-24   8.705662    8.705662    8.705662   9.075665  7.172425   
29  2018-05-23   8.737613    8.737613    8.737613   9.097172  7.191429   
30  2018-05-22   8.762646    8.762646    8.762646   9.083643  7.163947   
31  2018-05-21   8.782323    8.782323    8.782323   9.096051  7.228388   
32  2018-05-20   8.805675    8.805675    8.805675   9.112728  7.252054   
33  2018-05-19   8.765615    8.765615    8.765615   9.108308  7.190676   
34  2018-05-18   8.705165    8.705165    8.705165   9.102755  7.122060   

    magPrice1  magPrice2  
2    6.848005   6.848005  
3    6.879356   6.879356  
4    6.913737   6.913737  
5    6.915723   6.915723  
6    6.900731   6.900731  
7    6.892642   6.892642  
8    6.857514   6.857514  
9    6.842683   6.842683  
10   6.897705   6.897705  
11   6.948897   6.948897  
12   6.943122   6.943122  
13   6.858565   6.858565  
14   6.891626   6.891626  
15   6.906755   6.906755  
16   6.883463   6.883463  
17   7.003065   7.003065  
18   7.008505   7.008505  
19   6.978214   6.978214  
20   6.913737   6.913737  
21   6.911747   6.911747  
22   6.924612   6.924612  
23   6.882437   6.882437  
24   7.009409   7.009409  
25   7.085064   7.085064  
26   7.113956   7.113956  
27   7.090077   7.090077  
28   7.172425   7.172425  
29   7.191429   7.191429  
30   7.163947   7.163947  
31   7.228388   7.228388  
32   7.252054   7.252054  
33   7.190676   7.190676  
34   7.122060   7.122060

In [68]:
result3 = sm.ols(formula = 'goldPrice ~ magPrice + magPrice1 + magPrice2+ chipPrice + chipPrice1 + chipPrice2', data = data3).fit()

result3.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              goldPrice   R-squared:                       0.439
Model:                            OLS   Adj. R-squared:                  0.401
Method:                 Least Squares   F-statistic:                     11.71
Date:                Thu, 21 Jun 2018   Prob (F-statistic):           0.000174
Time:                        08:55:49   Log-Likelihood:                 55.138
No. Observations:                  33   AIC:                            -104.3
Df Residuals:                      30   BIC:                            -99.79
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     12.6042      0.828     15.231      0.000      10.914      14.294
magPrice       0.0246      0.023      1.056      0.300      -0.023       0.072
magPrice1      0.0246      0.023      1.056      0.300      -0.023       0.072
magPrice2      0.0246      0.023      1.056      0.300      -0.023       0.072
chipPrice     -0.1522      0.032     -4.829      0.000      -0.217      -0.088
chipPrice1    -0.1522      0.032     -4.829      0.000      -0.217      -0.088
chipPrice2    -0.1522      0.032     -4.829      0.000      -0.217      -0.088
==============================================================================
Omnibus:                        4.059   Durbin-Watson:                   0.847
Prob(Omnibus):                  0.131   Jarque-Bera (JB):                2.809
Skew:                           0.690   Prob(JB):                        0.245
Kurtosis:                       3.374   Cond. No.                     7.86e+18
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The smallest eigenvalue is  2e-34. This might indicate that there are
strong multicollinearity problems or that the design matrix is singular.
"""

**결과요약**

* magPrice의 p-value는 모두 0.3으로 설명력이 없었다.
* 하지만 chipPrice의 p-value는 0.000으로 설명력이 있었다.
* **durbin watson 결과 0.847이 나왔다.**
* **시나리오를 수정하면서 '데이터 칩'의 아이템만 p값이 굉장히 낮게 나왔다**

# 4-4. 시나리오4
y = dprice(t) + dprice(t-1) + dprice(t-2)인 경우



 * 계속되는 실행에서 '데이터 칩' 아이템의 p-value가 굉장히 낮게 나와서 '데이터 칩'만 이용해서 분석해봤다.






In [69]:
set4 = { 
    'Date' : chipPrice.loc[2::,'Date'],
    'goldPrice': goldPrice.loc[2:34,'Price'],  
    'chipPrice' : chipPrice.loc[2:,'Price'],
    'chipPrice1' : chipPrice.loc[2:,'Price'],
    'chipPrice2' : chipPrice.loc[2:,'Price']
}

data4 = pd.DataFrame(set4)

data4

Date  chipPrice  chipPrice1  chipPrice2  goldPrice
2   2018-06-19   8.848653    8.848653    8.848653   9.053687
3   2018-06-18   8.849371    8.849371    8.849371   9.079092
4   2018-06-17   8.875567    8.875567    8.875567   9.104980
5   2018-06-16   8.811652    8.811652    8.811652   9.097172
6   2018-06-15   8.747193    8.747193    8.747193   9.099409
7   2018-06-14   8.695841    8.695841    8.695841   9.106090
8   2018-06-13   8.721276    8.721276    8.721276   9.103868
9   2018-06-12   8.650850    8.650850    8.650850   9.110520
10  2018-06-11   8.694502    8.694502    8.694502   9.159047
11  2018-06-10   8.701679    8.701679    8.701679   9.179881
12  2018-06-09   8.628913    8.628913    8.628913   9.152711
13  2018-06-08   8.575273    8.575273    8.575273   9.129130
14  2018-06-07   8.535622    8.535622    8.535622   9.209340
15  2018-06-06   8.573384    8.573384    8.573384   9.219300
16  2018-06-05   8.637817    8.637817    8.637817   9.192176
17  2018-06-04   8.682369    8.682369    8.682369   9.197255
18  2018-06-03   8.653122    8.653122    8.653122   9.222269
19  2018-06-02   8.613775    8.613775    8.613775   9.221280
20  2018-06-01   8.532870    8.532870    8.532870   9.203316
21  2018-05-31   8.537192    8.537192    8.537192   9.174713
22  2018-05-30   8.582232    8.582232    8.582232   9.180912
23  2018-05-29   8.614683    8.614683    8.614683   9.182969
24  2018-05-28   8.621553    8.621553    8.621553   9.213336
25  2018-05-27   8.676587    8.676587    8.676587   9.275191
26  2018-05-26   8.688622    8.688622    8.688622   9.281730
27  2018-05-25   8.647871    8.647871    8.647871   9.235033
28  2018-05-24   8.705662    8.705662    8.705662   9.075665
29  2018-05-23   8.737613    8.737613    8.737613   9.097172
30  2018-05-22   8.762646    8.762646    8.762646   9.083643
31  2018-05-21   8.782323    8.782323    8.782323   9.096051
32  2018-05-20   8.805675    8.805675    8.805675   9.112728
33  2018-05-19   8.765615    8.765615    8.765615   9.108308
34  2018-05-18   8.705165    8.705165    8.705165   9.102755

In [70]:
result4 = sm.ols(formula = 'goldPrice ~ chipPrice + chipPrice1 + chipPrice2', data = data4).fit()

result4.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              goldPrice   R-squared:                       0.418
Model:                            OLS   Adj. R-squared:                  0.399
Method:                 Least Squares   F-statistic:                     22.23
Date:                Thu, 21 Jun 2018   Prob (F-statistic):           4.85e-05
Time:                        09:01:07   Log-Likelihood:                 54.536
No. Observations:                  33   AIC:                            -105.1
Df Residuals:                      31   BIC:                            -102.1
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     12.8729      0.789     16.318      0.000      11.264      14.482
chipPrice     -0.1427      0.030     -4.715      0.000      -0.204      -0.081
chipPrice1    -0.1427      0.030     -4.715      0.000      -0.204      -0.081
chipPrice2    -0.1427      0.030     -4.715      0.000      -0.204      -0.081
==============================================================================
Omnibus:                        7.374   Durbin-Watson:                   0.849
Prob(Omnibus):                  0.025   Jarque-Bera (JB):                5.812
Skew:                           0.941   Prob(JB):                       0.0547
Kurtosis:                       3.828   Cond. No.                     1.63e+35
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The smallest eigenvalue is 2.81e-67. This might indicate that there are
strong multicollinearity problems or that the design matrix is singular.
"""

**결과요약**

* chipPrice의 p-value는 0.000으로 설명력이 있었다.
* **durbin watson 결과 0.849로 나타났다.**


# 4-5. 시나리오5
y = magprice(t-1) + magprice(t-2) + dprice(t-1) + dprice(t-2)인 경우

In [73]:
set5 = { 
    'Date' : chipPrice.loc[2::,'Date'],
    'goldPrice': goldPrice.loc[2:34,'Price'],
    'magPrice1' : magPrice.loc[2:,'Price'],
    'magPrice2' : magPrice.loc[2:,'Price'],
    'chipPrice1' : chipPrice.loc[2:,'Price'],
    'chipPrice2' : chipPrice.loc[2:,'Price']
}

data5 = pd.DataFrame(set5)

data5

Date  chipPrice1  chipPrice2  goldPrice  magPrice1  magPrice2
2   2018-06-19    8.848653    8.848653   9.053687   6.848005   6.848005
3   2018-06-18    8.849371    8.849371   9.079092   6.879356   6.879356
4   2018-06-17    8.875567    8.875567   9.104980   6.913737   6.913737
5   2018-06-16    8.811652    8.811652   9.097172   6.915723   6.915723
6   2018-06-15    8.747193    8.747193   9.099409   6.900731   6.900731
7   2018-06-14    8.695841    8.695841   9.106090   6.892642   6.892642
8   2018-06-13    8.721276    8.721276   9.103868   6.857514   6.857514
9   2018-06-12    8.650850    8.650850   9.110520   6.842683   6.842683
10  2018-06-11    8.694502    8.694502   9.159047   6.897705   6.897705
11  2018-06-10    8.701679    8.701679   9.179881   6.948897   6.948897
12  2018-06-09    8.628913    8.628913   9.152711   6.943122   6.943122
13  2018-06-08    8.575273    8.575273   9.129130   6.858565   6.858565
14  2018-06-07    8.535622    8.535622   9.209340   6.891626   6.891626
15  2018-06-06    8.573384    8.573384   9.219300   6.906755   6.906755
16  2018-06-05    8.637817    8.637817   9.192176   6.883463   6.883463
17  2018-06-04    8.682369    8.682369   9.197255   7.003065   7.003065
18  2018-06-03    8.653122    8.653122   9.222269   7.008505   7.008505
19  2018-06-02    8.613775    8.613775   9.221280   6.978214   6.978214
20  2018-06-01    8.532870    8.532870   9.203316   6.913737   6.913737
21  2018-05-31    8.537192    8.537192   9.174713   6.911747   6.911747
22  2018-05-30    8.582232    8.582232   9.180912   6.924612   6.924612
23  2018-05-29    8.614683    8.614683   9.182969   6.882437   6.882437
24  2018-05-28    8.621553    8.621553   9.213336   7.009409   7.009409
25  2018-05-27    8.676587    8.676587   9.275191   7.085064   7.085064
26  2018-05-26    8.688622    8.688622   9.281730   7.113956   7.113956
27  2018-05-25    8.647871    8.647871   9.235033   7.090077   7.090077
28  2018-05-24    8.705662    8.705662   9.075665   7.172425   7.172425
29  2018-05-23    8.737613    8.737613   9.097172   7.191429   7.191429
30  2018-05-22    8.762646    8.762646   9.083643   7.163947   7.163947
31  2018-05-21    8.782323    8.782323   9.096051   7.228388   7.228388
32  2018-05-20    8.805675    8.805675   9.112728   7.252054   7.252054
33  2018-05-19    8.765615    8.765615   9.108308   7.190676   7.190676
34  2018-05-18    8.705165    8.705165   9.102755   7.122060   7.122060

In [74]:
result5 = sm.ols(formula = 'goldPrice ~ magPrice1 + magPrice2 + chipPrice1 + chipPrice2', data = data5).fit()

result5.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              goldPrice   R-squared:                       0.439
Model:                            OLS   Adj. R-squared:                  0.401
Method:                 Least Squares   F-statistic:                     11.71
Date:                Thu, 21 Jun 2018   Prob (F-statistic):           0.000174
Time:                        09:12:27   Log-Likelihood:                 55.138
No. Observations:                  33   AIC:                            -104.3
Df Residuals:                      30   BIC:                            -99.79
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     12.6042      0.828     15.231      0.000      10.914      14.294
magPrice1      0.0369      0.035      1.056      0.300      -0.034       0.108
magPrice2      0.0369      0.035      1.056      0.300      -0.034       0.108
chipPrice1    -0.2283      0.047     -4.829      0.000      -0.325      -0.132
chipPrice2    -0.2283      0.047     -4.829      0.000      -0.325      -0.132
==============================================================================
Omnibus:                        4.059   Durbin-Watson:                   0.847
Prob(Omnibus):                  0.131   Jarque-Bera (JB):                2.809
Skew:                           0.690   Prob(JB):                        0.245
Kurtosis:                       3.374   Cond. No.                     2.55e+18
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The smallest eigenvalue is 1.27e-33. This might indicate that there are
strong multicollinearity problems or that the design matrix is singular.
"""

**결과요약**

* chipPrice의 p-value는 0.000으로 설명력이 있었다.
* 하지만 magPrice의 p-value는 0.3으로 설명력이 낮았다.
* **durbin watson 결과 0.847로 나타났다.**

* **durbin watson 결과가 시나리오에 상관없이 일정한 수준으로 나타난다.**

# 5. 결론


![대체 텍스트](https://postfiles.pstatic.net/MjAxODA2MjFfMjcz/MDAxNTI5NTc0MDAyMTI5.V0Tr1Wybxy-ibPeXrShUVvGfNDIMeX3pb0n4m-a7ZjQg.CM0eeB2ivkAIiNVFkXzdbDAHqXNQTms29HG-vgSVUN8g.PNG.bububu111/%EC%BA%A1%EC%B2%98.PNG?type=w773)


*  모든 시나리오에서 '데이터 칩'의 p-value는 0에 수렴하는 값이 도출됬다.
*  여러 조합으로 시나리오를 실행했을때, 시나리오 4번이 가장 의미있는 식으로 예상된다.
*  그런데 데이터의 양이 너무 적어서(약 30일) 유의미한 결과를 도출하지는 못했다.
*  과거 아이템 가격에 대한 데이터를 더 모을 수 있으면 더 의미있는 결과를 나타냈을 것 같다.
